airbyte-cdk/bulk/core/base/src/main/kotlin/io/airbyte/cdk/spec/SpecOperation.kt

```diff
@@ -15,19 +15,30 @@ import java.net.URI
 @Requires(property = Operation.PROPERTY, value = "spec")
 class SpecOperation(
     @Value("\${airbyte.connector.metadata.documentation-url}") val documentationUrl: String,
-    val configJsonObjectSupplier: ConfigurationSpecificationSupplier<*>,
-    val extendSpecification: SpecificationExtender,
+    val specificationFactory: SpecificationFactory,
     val outputConsumer: OutputConsumer,
 ) : Operation {
     override fun execute() {
-        val spec =
-            ConnectorSpecification()
-                .withDocumentationUrl(URI.create(documentationUrl))
-                .withConnectionSpecification(configJsonObjectSupplier.jsonSchema)
-        outputConsumer.accept(extendSpecification(spec))
+        val spec = specificationFactory.create().withDocumentationUrl(URI.create(documentationUrl))
+        outputConsumer.accept(spec)
     }
 }
 
+interface SpecificationFactory {
+    fun create(): ConnectorSpecification
+}
+
+@Singleton
+class ConfigurationSupplierSpecificationFactory(
+    val configJsonObjectSupplier: ConfigurationSpecificationSupplier<*>,
```

Comment:
Is there any way to just inject the `jsonSchema` directly instead of doing this lazy stuff?